src/main/java/frc/robot/sensors/IrSensor.java

```diff
@@ -0,0 +1,66 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+/**
+ * Code by Charlie Macdonald.
+ * Last modified on Feb 5th, 2020.
+ */
+
+package frc.robot.sensors;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.MedianFilter;
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+public class IrSensor implements Sendable {
+    private static final int kIRPort = 1; //set IR sensor port
+        //TODO: The IR sensor's port may change, depending on the robot setup. (Use a robot-specific config to set this?)
```

Comment:
Yes, the channel for the sensor should be defined in the Config.java file with a robotSpecific value. This way it can automatically switch between channels based on which robot it's deployed to

Reply:
I think this should be a value passed into the constructor so I can do this:

```
IrSensor inputSensor = new IrSensor(0); // input sensor is connected to analog input 0
IrSensor outputSensor = new IrSensor(2); // output sensor is connected to analog input 2;
```

Reply:
I think this should be given in the constructor (see my other comment)

Reply:
Yup, fixed that. Channel number can now be passed as an argument.